[Site Design Revamp] Remove preview mode selector from nav bar

twstokes opened the pull request:
- Item **A** Subtask of #18433
- Related PR: https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/309

**This PR:**
- Removes the mode selector from the nav bar on the Site Design screen
- Enforces mobile-sized thumbnails for all devices

| Before - iPhone | After - iPhone |
| - | - |
| ![Simulator Screen Shot - iPhone 13 - 2022-04-27 at 17 49 49](https://user-images.githubusercontent.com/2092798/165637044-a9a9f33e-766c-4c66-a85d-b6e90ba64afa.png) | ![image](https://user-images.githubusercontent.com/2092798/165621913-96c628f7-2641-421f-be5b-c5f8cf58c03f.png) |

| Before - iPad | After - iPad |
| - | - |
| ![Simulator Screen Shot - iPad (9th generation) - 2022-04-27 at 16 02 07](https://user-images.githubusercontent.com/2092798/165621572-be63f504-1390-497b-9299-d88c71fafa00.png) | ![image](https://user-images.githubusercontent.com/2092798/165621956-892d49c6-230c-4d64-8cca-da05021cf2be.png) |

## To test:

### Mobile thumbnails - Phone
1. Using a phone, start the Site Creation Flow and navigate to the Site Design screen
2. Expect thumbnails to be screenshots of the mobile layout
3. Select a design and tap the "Preview" button
4. Expect preview to be in the mobile layout
5. Tap the device mode picker at the top right and confirm that the mode is set to Mobile
6. Tap Tablet to switch to the tablet mode
7. Expect preview to be in the tablet layout
8. Dismiss the preview
9. Expect thumbnails to be screenshots of the mobile layout (they did not switch to tablet layout)
10. Select a design and tap the "Preview" button
11. Expect preview to be in the tablet layout (the Preview view retained the last layout mode)

### Mobile thumbnails - iPad
1. Using an iPad, start the Site Creation Flow and navigate to the Site Design screen
2. Expect thumbnails to be screenshots of the mobile layout (not tablet)
3. Select a design and tap the "Preview" button
4. Expect preview to be in the tablet layout
5. Tap the device mode picker at the top right and confirm that the mode is set to Tablet
6. Tap Mobile to switch to the mobile mode
7. Expect preview to be in the mobile layout
8. Dismiss the preview
9. Expect thumbnails to be screenshots of the mobile layout (they did not switch to tablet layout)
10. Select a design and tap the "Preview" button
11. Expect preview to be in the mobile layout (the Preview view retained the last layout mode)

### Tracks events - Phone
1. Using a phone, start the Site Creation Flow and navigate to the Site Design screen
12. Expect `enhanced_site_creation_site_design_viewed` to be fired with `preview_mode` set to `mobile`
13. Select a design and tap the "Preview" button
14. Expect `enhanced_site_creation_site_design_preview_viewed` to be fired with `preview_mode` set to `mobile` and `template` set to the corresponding design
15. Expect `enhanced_site_creation_site_design_preview_loading` to be fired with `preview_mode` set to `mobile` and `template` set to the corresponding design
16. Expect `enhanced_site_creation_site_design_preview_loaded` to be fired with `preview_mode` set to `mobile` and `template` set to the corresponding design
17. Tap the device mode picker at the top right and switch to Tablet
18. Expect `enhanced_site_creation_site_design_preview_mode_button_tapped` to be fired with `preview_mode` set to `mobile`
19. Expect `enhanced_site_creation_site_design_preview_mode_changed ` to be fired with `preview_mode` set to `tablet`
20. Expect `enhanced_site_creation_site_design_preview_loading` to be fired with `preview_mode` set to `tablet` and `template` set to the corresponding design
21. Expect `enhanced_site_creation_site_design_preview_loaded` to be fired with `preview_mode` set to `tablet` and `template` set to the corresponding design
22. Tap "Create Site" (the button may say "Choose" depending on Site Name treatment)
23. Expect `enhanced_site_creation_site_design_selected` to be fired with `template` set to the corresponding design

### Tracks events - iPad
1. Using an iPad, start the Site Creation Flow and navigate to the Site Design screen
2. Expect `enhanced_site_creation_site_design_viewed` to be fired with `preview_mode` set to `mobile`
3. Select a design and tap the "Preview" button
4. Expect `enhanced_site_creation_site_design_preview_viewed` to be fired with `preview_mode` set to `mobile` and `template` set to the corresponding design
5. Expect `enhanced_site_creation_site_design_preview_loading` to be fired with `preview_mode` set to `mobile` and `template` set to the corresponding design
6. Expect `enhanced_site_creation_site_design_preview_loaded` to be fired with `preview_mode` set to `mobile` and `template` set to the corresponding design
7. Tap the device mode picker at the top right and switch to Mobile
8. Expect `enhanced_site_creation_site_design_preview_mode_button_tapped` to be fired with `preview_mode` set to `tablet`
9. Expect `enhanced_site_creation_site_design_preview_mode_changed ` to be fired with `preview_mode` set to `mobile`
12. Expect `enhanced_site_creation_site_design_preview_loading` to be fired with `preview_mode` set to `mobile` and `template` set to the corresponding design
13. Expect `enhanced_site_creation_site_design_preview_loaded` to be fired with `preview_mode` set to `mobile` and `template` set to the corresponding design
14. Tap "Create Site" (the button may say "Choose" depending on Site Name treatment)
15. Expect `enhanced_site_creation_site_design_selected` to be fired with `template` set to the corresponding design

## Regression Notes
1. Potential unintended areas of impact
    - Site Design thumbnails
    - Tracks event sent related to the Site Design view

2. What I did to test those areas of impact (or what existing automated tests I relied on)
    - Manual steps above

3. What automated tests I added (or what prevented me from doing so)
    - Added `testSiteDesignPreviewDeviceIsAlwaysMobile`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

twstokes committed:
Remove preview mode selector from the nav bar.

twstokes committed:
Remove Tracks helper function for the thumbnail preview mode.

twstokes committed:
Point to WordPressShared pod branch.

twstokes committed:
Remove Tracks event.

twstokes committed:
Enforce mobile thumbnails and prevent changing them with the device mode picker for previews.

twstokes committed:
Override the getter and setter for the superclass property.

twstokes committed:
Test that the preview device can't be changed on design screen.

twstokes committed:
Add test for preview view.

mkevins commented on WordPress/WordPressTest/SiteCreation/SiteDesignTests.swift:
```diff
@@ -73,4 +73,42 @@ class SiteDesignTests: XCTestCase {
+        XCTAssertEqual(siteDesignVC.selectedPreviewDevice, expectedDevice)
+    }
+
+    /// Tests that the preview device on the Preview view can be changed
+    func testSiteDesignPreviewPreviewDeviceCanBeChanged() throws {
```
This comment makes it clear that the `PreviewPreview` in the function name is intentional :smile: .

ovitrif replied:
I could suggest `PreviewPreview` → `PreviewViewPreview` 😅

twstokes replied:
I cringed a little when I made `testSiteDesignPreviewPreviewDeviceCanBeChanged` so I'll tweak that to a better name. 😄

twstokes replied:
I ended up removing this test altogether after the last batch of changes as it seemed to lose its value.

mkevins commented on WordPress/WordPressTest/SiteCreation/SiteDesignTests.swift:
```diff
@@ -73,4 +73,42 @@ class SiteDesignTests: XCTestCase {
+        siteDesignPreviewVC.loadViewIfNeeded()
+        siteDesignPreviewVC.viewDidLoad()
+        XCTAssertEqual(siteDesignPreviewVC.selectedPreviewDevice, PreviewDeviceSelectionViewController.PreviewDevice.mobile)
+        siteDesignPreviewVC.selectedPreviewDevice = PreviewDeviceSelectionViewController.PreviewDevice.tablet
+
+        // then
+        XCTAssertEqual(siteDesignPreviewVC.selectedPreviewDevice, expectedDevice)
```
Is this just essentially asserting that there's not a noop setter on the preview vc?

twstokes replied:
This one is asserting that on the Preview VC we can change the mode. On the main design VC we do the no op. 👍

twstokes replied:
I ended up removing this test altogether after the last batch of changes as it seemed to lose its value.

mkevins reviewed (approved):
The code changes look good, and this works as expected (tested on iPhone 11, and iPad Pro (12.9-inch) simulator). I had a question about the added test, but it isn't a blocker.

One subtle thing I noticed, that could be different from the Android implementation, is that the selected preview mode in the preview screen gets reset every time it is re-opened (on Android, this is "remembered" when back is pressed, since the underlying state is persisted in the thumbnail screen). I think either behavior is fine, fwiw, and I'm just noting this for completeness.

twstokes committed:
Point to feature branch for iOS Shared pod.

twstokes commented:
> The code changes look good, and this works as expected (tested on iPhone 11, and iPad Pro (12.9-inch) simulator). I had a question about the added test, but it isn't a blocker.
> 
> One subtle thing I noticed, that could be different from the Android implementation, is that the selected preview mode in the preview screen gets reset every time it is re-opened (on Android, this is "remembered" when back is pressed, since the underlying state is persisted in the thumbnail screen). I think either behavior is fine, fwiw, and I'm just noting this for completeness.

Thanks @mkevins - that's a good observation and a pattern I hadn't considered. I imagine users will want to stick to the preview mode as they try out different designs, so I'm going to update this PR to match that functionality.

ovitrif commented:
Just a remark that I can review the additional updates when they are pushed 🙇

twstokes committed:
Store the selected preview device for the Preview view.

twstokes committed:
Remove test for Preview view.

twstokes committed:
Tweak access control for Preview view device.

twstokes committed:
Use device default for initial Preview device size.

twstokes commented:
> Just a remark that I can review the additional updates when they are pushed 🙇

Thank you @ovitrif! This is ready for review again. 

The changes include:
- The Preview view will retain the last mode it was in.
- I removed the `testSiteDesignPreviewPreviewDeviceCanBeChanged` test as I didn't find as much value in it as originally thought.
- I added steps 10 and 11 to the Thumbnails tests.

twstokes committed:
Reword title.

twstokes committed:
Hide prompt view and remove text.

twstokes committed:
Make the prompt text optional and hide the view when it's nil.

ovitrif commented on WordPress/Classes/ViewRelated/Site Creation/Design Selection/SiteDesignContentCollectionViewController.swift:
```diff
@@ -136,7 +138,15 @@ class SiteDesignContentCollectionViewController: FilterableCategoriesViewControl
         guard let design = selectedDesign else { return }
 
-        let previewVC = SiteDesignPreviewViewController(siteDesign: design, selectedPreviewDevice: PreviewDevice.default, createsSite: createsSite, onDismissWithDeviceSelected: nil, completion: completion)
+        let previewVC = SiteDesignPreviewViewController(
```
Nice, I love params on separate lines 👍 , so much better in diffs.

ovitrif reviewed (approved):
Great work @twstokes ! 🎉 

Works as expected in tests. Let's :shipit:

ovitrif commented:
A few CI checks failed, seems nothing caused by the changes, thus I retriggered them 🤞

ovitrif committed:
Merge pull request #18474 from wordpress-mobile/task/site-design-main-screen-ui

[Site Design Revamp] Update title text and remove subtext

twstokes commented:
> A few CI checks failed, seems nothing caused by the changes, thus I retriggered them 🤞

Thanks @ovitrif for the review and for retriggering the tests! Looks like they're all good now. ✅
